### DISCUSSION TITLE
NSClient++ Export Control Classification Number (ECCN)

### DISCUSSION THREAD

**Discussion by jmsanesteban** [mickem/nscp#804] [Q&A]
Good morning all,

We need to know the ECCN for the agent due to some regulations. I tried to figure out how to get that code, but everything points me to the developers of the product. 

I reached: https://www.trade.gov/how-do-i-determine-my-export-control-classification-number-eccn -> https://www.bis.doc.gov/index.php/regulations/commerce-control-list-ccl -> https://www.bis.doc.gov/index.php/documents/regulations-docs/2329-commerce-control-list-index-3/file and I didn’t find anything useful.

From our company department:
'In any case, if the software was not developed internally, you will have to get in contact with the provider to get the correct ECCN of their product because we at this department are not to do so. There are many other parameters that come into consideration such as the type of encryption, the amount of US content, etc., and the functionalities used… that only the developer is able to know.'

Have you any idea of the number or how to get it?

Thanks in advance.

BR,
Juanma.

**Comment by mickem** [comment]
I am no expert but since this is not a US product and it is not shipped I do not think there are any such numbers...

**Comment by mickem** [comment]
To add a bit to the earlier answer, for whoever needs to fill in the form (this is not legal advice; your trade-compliance people have the last word):

- NSClient++ is open-source software (GPL-2.0), developed in Sweden with no US origin, and the source and binaries are publicly available to anyone at no cost from GitHub. Nothing is "shipped" or sold.
- The only cryptography it contains is the publicly available OpenSSL library, used for TLS on the NRPE and web listeners.

The project therefore has never been classified and has no ECCN to hand out. In practice, companies that need an entry for it treat it as *publicly available software*, which under the US EAR is not subject to the regulations (15 CFR 734.3(b)(3) and 734.7); published open-source encryption code only carries the notification requirement of 742.15(b), which falls on whoever publishes it, not the user. Many organisations record such software as "not subject to the EAR" or as EAR99 in their systems. If your department insists on a classification from the developer, the honest answer is that there is none.
